test/test_functional.py

```diff
@@ -0,0 +1,184 @@
+import math
+
+import torch
+import torchaudio
+import unittest
+import test.common_utils
+
+
+class TestFunctional(unittest.TestCase):
+    data_sizes = [(2, 20), (3, 15)]
+    number_of_trials = 100
+
+    def _compare_estimate(self, sound, estimate, atol=1e-6, rtol=1e-8):
+        # trim sound for case when constructed signal is shorter than original
+        sound = sound[..., :estimate.size(-1)]
+
+        self.assertTrue(sound.shape == estimate.shape, (sound.shape, estimate.shape))
+        self.assertTrue(torch.allclose(sound, estimate, atol=atol, rtol=rtol))
+
+    def _test_istft_is_inverse_of_stft(self, kwargs):
+        # generates a random sound signal for each tril and then does the stft/istft
+        # operation to check whether we can reconstruct signal
+        for data_size in self.data_sizes:
+            rtg = test.common_utils.RandomTensorGenerator(seed=0, size=data_size)
+            for _ in range(self.number_of_trials):
+                sound = rtg.rand_float_tensor()
```

Comment:
This suffers from the same issue has using a regular number generator, which is that its stateful and the results dependent on the number of previous calls to rand_float_tensor